client/src/components/CosmologyForm.vue

```diff
@@ -13,90 +13,138 @@
       </md-select>
     </md-field>
 
-    <FormNumberField
+    <DoubleField
       v-for="(input, inputName) in inputs"
-      :labelHtml="input.html"
+      :htmlParam="input.html"
       :key="inputName"
       :step="input.step"
-      :currentValue="cosmoValues[inputName]"
+      :value="cosmoData.cosmo_params[inputName] !== undefined
+        ? cosmoData.cosmo_params[inputName]
+        : cosmoData[inputName]"
+      v-on:input="createSetCurrentValueFunc(inputName)($event)"
```

Comment:
The logic you use for :value could be used for v-model and you wouldn't need your createSetCurrentValueFunc function at all because it would connect everything without any trouble -- you just might need to watch the object that contains it and update your allCosmoData to support that.

Reply:
How might that be implemented in less code? Having a hard time visualizing that.

Reply:
You would have

v-model="cosmoData.cosmo_params[inputName] !== undefined
        ? cosmoData.cosmo_params[inputName]
        : cosmoData[inputName]"

instead of :value & v-on:input. This connects the DoubleField to whatever field in cosmoData needs it. 

then down below you could have 

watch: {
  cosmoData: {
    deep=True,
    handler() {
       this.allCosmoData[this.cosmoData.cosmo_model] = this.cosmoData;
       this.$emit('updateCosmo', this.cosmoData);
    }, 
  }
}

Reply:
It's quite possible that v-model won't even accept a computed property like that, though -- I'm not completely sure! I will go ahead and approve but if you feel like tinkering and figure out if it works lemme know haha